quickwit/quickwit-serve/src/elasticsearch_api/bulk_v2.rs

```diff
@@ -293,13 +300,25 @@ fn make_elastic_bulk_response_v2(
                 error: Some(error),
             };
             let action = ElasticBulkAction::Index(item);
-            actions.push(action);
+            positioned_actions.push((doc_handle.doc_position, action));
         }
     }
     assert!(
         per_subrequest_doc_handles.is_empty(),
         "doc handles should be empty"
     );
+
+    assert_eq!(
+        positioned_actions.len(),
+        action_count,
+        "request and response action count should match"
+    );
+    positioned_actions.sort_unstable_by_key(|(idx, _)| *idx);
```

Comment:
instead of sorting, we could have an array of `Option<Action>` where we write the result at the correct position directly. that would be faster, I'm not sure how much we care though, here is probably not performance critical

Reply:
Yes, I went that way originally, but the code was a bit confused and as you say, I doubt the performance gain is significative.